doc/api/stream.md

```diff
@@ -443,9 +443,26 @@ first argument. To reliably detect write errors, add a listener for the
 The return value is `true` if the internal buffer does not exceed
 `highWaterMark` configured when the stream was created after admitting `chunk`.
 If `false` is returned, further attempts to write data to the stream should
-stop until the [`'drain'`][] event is emitted. However, the `false` return
-value is only advisory and the writable stream will unconditionally accept and
-buffer `chunk` even if it has not not been allowed to drain.
+stop until the [`'drain'`][] event is emitted.
+
+In case the `false` return value is ignored, the writable stream will
+unconditionally accept and buffer `chunk` even if it has not not been allowed
+to drain. However, this might lead to memory leaks and garbage
```

Comment:
> Also, why is this especially true for Transform streams? If we are going to say that its more true for particular derived types of Writable, we should say why.

A `Writable` in Node core is always writing thing to the underlining resource, e.g. the socket. You can stop that with `cork()`, but by default it flows the data. `Transform` things are paused by default until they are piped or an `'data'` or `'readable'` event handler is added.
So, ignoring when `write()` returns false will cause the data to be buffered until the stream is flowing. If that does not happen, you have a memory leak.

Should I add the above to the PR?